### DISCUSSION TITLE
Browser to dump an entire SILO file?

### DISCUSSION THREAD

**Discussion by wnissen** [LLNL/Silo#422] [Q&A] [answered]
Hi, @markcmiller86 et al.,

I can use browser to print the value of a particular variable, e.g.:
```
> browser --eval=MMESH Disk2D/OvlTop.silo
INFO: opening `Disk2D/OvlTop.silo' as $1
                              = DBmultimesh: struct
id                            =    0
[...]
groupnames                    =    NULL
```
Is there a command or recipe to (hopefully recursively) dump all the variables contained within a given SILO file?

Thanks!

**Comment by markcmiller86** [comment]
@wnissen I think you might be able to diff the file with itself and have it print everything, not just diffs. Lemme try.

**Comment by markcmiller86** [comment, marked as answer]
@wnissen ok, that doesn't work. When the file is diffed with itself, all the diffs are zero and I cannot set a negative tolerance. If it is a Silo/HDF5 file you can use HDF5 tools (`h5ls` and `h5dump`). But, you wind up having to understand a bit about the *internals* of how Silo writes its high level objects to HDF5.

Another option is to use Silo's python interface and write a small amount of python code to do it....recursively using `Silo.SetDir()` and `Silo.GetToc()`.

**Reply by wnissen** [reply]
Thanks, Mark, I figured if anyone knew the shortcut, it would be you. I can write something fairly straightforward in Python.

**Reply by cyrush** [reply]
@wnissen  you may be able to use Conduit's Silo / Overlink support to read in two files and do a full diff of fields.

**Reply by wnissen** [reply]
@cyrush Do I do that with the Conduit Python? Is there an example of reading and/or diffing using that? Thanks.

**Reply by wnissen** [reply]
Looks like this is quite easy to dump the Conduit, that is just as good if not better than the `browser` output.
```
>>> import conduit
>>> node = conduit.Node()
>>> conduit.relay.io.silo.read_mesh(node, 'Disk2D.silo')
>>> node.print_detailed()

{
  "domain_000000": 
  {
    "topologies": 
    {
      "MMESH": 
      {
        "elements": 
        {
          "shape": 
          {
            "dtype":"char8_str",
            "number_of_elements": 4,
            "offset": 0,
            "stride": 1,
            "element_bytes": 1,
            "endianness": "little",
            "value": "hex"
          },
          "connectivity": 
          {
[etc.]
